### PR TITLE
fix: remove [warning] from typedoc for external usage

### PR DESCRIPTION
**Pull request by 0xknwn**
running `npx typedoc --entryPoints src --out build` generates a number of warnings that are addressed here.  This PR replaces #1092. It is based on `next-version` and the changes are now split by commits so that they are easy to understand and if needed can be cherry-picked

## Motivation and Resolution

Some of the issues addressed that are small enhancements to use the starknet.js library. Others are more opinionated or improvements. For a better understanding, changes have been grouped per commits. This is the list of commits and fixes:

- fix: rename the @param to match the definitions
- fix: add a space before typedoc \<br/\> so that it catches the name
- fix: improve function declaration so that typedoc matches it
- fix: reintroduce the parameter name
- fix: remove unused type declaration and export
- ~~fix: export underlying types for RPC 0.6~~
- ~~fix: export the underlying types with the composite types~~

### RPC version (if applicable)

0.6 and 0.7

## Usage related changes

None

## Development related changes

None

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Comment by 0xknwn** [comment]
I will rebase the PR but for now `next-version` seems broken. When I run npm run build, I got `✘ [ERROR] Could not resolve "starknet-types-07"`. Let me know when it is fixed.

**Comment by 0xknwn** [comment]
I've rebased. During the process, I've hit an issue I had to workaround with PR #1098. In order pass this PR, I had to cherry pick the commit from #1098. I will remove it once we agree on a fix for this other issue.

File: src/types/api/rpcspec_0_6/methods.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ import {
 
 export type Methods = ReadMethods & WriteMethods & TraceMethods;
 
-type ReadMethods = {
+export type ReadMethods = {
```
Comment (PhilippeR26) [inline src/types/api/rpcspec_0_6/methods.ts:44]: Same remark, related to previous comment.

**Review by PhilippeR26 — COMMENTED** [review]
My feeling is that there are exports everywhere, and I am not convinced that they are all useful for users.

**Comment by 0xknwn** [comment]
Thank you @PhilippeR26 for your quick review!

> My feeling is that there are exports everywhere, and I am not convinced that they are all useful for users.

First thing first: This PR is split by issues: each one being in a commit. The `export` are in the 2 commits that are `[3354807](https://github.com/starknet-io/starknet.js/pull/1095/commits/3354807a7295d929f59e0a2d5e566e10bd0306e1)` and [7587417](https://github.com/starknet-io/starknet.js/pull/1095/commits/7587417105e1ffe9782ca5262e702d5438928a65).

I will remove them from this PR so that we discuss my proposal later if you do not mind!

**Comment by 0xknwn** [comment]
> Let @tabaktoni decide what is the best way to proceed.

@PhilippeR26 your comment was actually very relevant and it took me 3 min to remove the commits. Check for yourself. The PR is way simpler and can be reviewed peacefully... No disagreements, hopefully 🤣.

Once done, we can have a chat: my opinion is that `typedoc` should not return any error or warning but I confess I am more often wrong than right.

**Review by tabaktoni — APPROVED** [review]
Mostly cleanups, lgtm

**Review by PhilippeR26 — COMMENTED** [review]
We have still a problem to build the Docusaurus site :
image
The API doc for Next doc version is not generated.

**Comment by raizo07** [comment]
> We have still a problem to build the Docusaurus site : image The API doc for Next doc version is not generated.

Hello @PhilippeR26 I can build the Docusaurus site if need be and it is an issue. I just built that of **Unruggable Meme** last week with inspiration from [@uniswap/docs](https://github.com/Uniswap/docs) as requested.

**Comment by PhilippeR26** [comment]
Sorry, I didn't understood your response.

**Comment by PhilippeR26** [comment]
Sorry, still not clear.
We have today a documentation site (https://www.starknetjs.com), generated by Docusaurus.
I was working fine until v6.7.0 (just some non blocking warnings).
You have solved these warnings.
But from v6.8.0, we have some blocking errors, that have to be solved (probably related with #1098).

**Comment by raizo07** [comment]
> Sorry, still not clear. We have today a documentation site (https://www.starknetjs.com), generated by Docusaurus. I was working fine until v6.7.0 (just some non blocking warnings). You have solved these warnings. But from v6.8.0, we have some blocking errors, that have to be solved (probably related with #1098).

Oh alright, I understand your point now.

**Comment by ivpavici** [comment]
@0xknwn can i ask you please to rebase and resolve the conflict, and we can merge!

**Comment by 0xknwn** [comment]
> @0xknwn can i ask you please to rebase and resolve the conflict, and we can merge!

done

**Pull request merged**